features/steps/permissions_component.go

```diff
@@ -0,0 +1,154 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	dpMongoDriver "github.com/ONSdigital/dp-mongodb/v2/mongodb"
+	"github.com/ONSdigital/dp-permissions-api/api"
+	"github.com/ONSdigital/dp-permissions-api/config"
+	"github.com/ONSdigital/dp-permissions-api/mongo"
+	"github.com/ONSdigital/dp-permissions-api/service"
+	serviceMock "github.com/ONSdigital/dp-permissions-api/service/mock"
+	"github.com/benweissmann/memongo"
+	"github.com/cucumber/godog"
+	"github.com/gofrs/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"net/http"
+)
+
+type PermissionsComponent struct {
+	ErrorFeature   componenttest.ErrorFeature
+	svc            *service.Service
+	errorChan      chan error
+	MongoClient    *mongo.Mongo
+	Config         *config.Config
+	HTTPServer     *http.Server
+	ServiceRunning bool
+}
+
+func NewPermissionsComponent(mongoFeature *componenttest.MongoFeature) (*PermissionsComponent, error) {
+
+	f := &PermissionsComponent{
+		HTTPServer:     &http.Server{},
```

Comment:
Im not 100% sure but I think there is is there a dp variant of this we should be using. I think it will be in `dp-net`

https://github.com/ONSdigital/dp-net/blob/main/http/server.go

Reply:
Updated the pr to use dp-net http server